src/InputGroup/InputGroup.js

```diff
@@ -1,272 +1,79 @@
-import Button from '../Button/Button';
 import classnames from 'classnames';
-import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
 import FormInput from '../Forms/FormInput';
-import Icon from '../Icon/Icon';
+import { INPUT_GROUP_ADDON_TYPES } from '../utils/constants';
+import InputGroupAddon from './_InputGroupAddon';
 import PropTypes from 'prop-types';
 import withStyles from '../utils/WithStyles/WithStyles';
-import { INPUT_GROUP_ADDON_POSITIONS, INPUT_GROUP_TYPES } from '../utils/constants';
 import React, { Component } from 'react';
-
+/*eslint-disable*/
 class InputGroup extends Component {
     constructor(props) {
         super(props);
-
-        this.state = {
-            value: this.props.inputValue,
-            searchValue: this.props.inputValue
-        };
     }
 
-    handleUp = e => {
-        e.preventDefault();
-        this.setState({
-            value: parseInt(this.state.value, 10) + 1
-        });
-    };
-
-    handleDown = e => {
-        e.preventDefault();
-        this.setState({
-            value: parseInt(this.state.value, 10) - 1
-        });
-    };
-
-    handleClear = e => {
-        e.preventDefault();
-        this.setState({
-            searchValue: ''
-        });
-    };
-
-    handleChange = e => {
-        e.preventDefault();
-        this.setState({
-            searchValue: e.target.value
-        });
-    };
-
-    handleTextChange = e => {
-        e.preventDefault();
-        this.setState({
-            value: e.target.value
-        });
-    };
-
     render() {
-        const {
+        let {
             actions,
-            addonClassNames,
-            addonPos,
-            addon,
+            addon, 
             children,
             className,
             compact,
             disableStyles,
-            glyph,
-            inputClassName,
-            inputType,
-            inputId,
-            inputName,
-            inputPlaceholder,
-            inputProps,
-            inputValue,
-            localizedText,
-            numberDownButtonProps,
-            numberUpButtonProps,
             ...props
         } = this.props;
 
-        const inputGroupClasses = classnames(
-            className,
-            'fd-input-group'
-        );
-
-        const addonClasses = classnames(
-            addonClassNames,
-            'fd-input-group__addon',
-            [{ 'fd-input-group__addon--button': !!actions || inputType === 'number' }]
-        );
-
         const inputClasses = classnames(
```

Comment:
nit: this probably doesn't need to call `classnames` anymore if it's one static string.